common_components/monitoring/json_to_dot.py

```diff
@@ -0,0 +1,285 @@
+#!/usr/bin/python3
+
+# Copyright 2022 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     https://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Python script for converting a list of performance nodes into a DOT graph.
+
+   Expected list of nodes is in JSON of the format:
+     - List of objects
+     - Each object having the following:
+        - name - name of the node
+        - depends_on - list of upstream dependencies (correponding to other
+          nodes)
+        - start_time - Javascript formatted timestamp when node started
+        - end_time - Javascript formatted timestamp when node ended
+"""
+
+import os.path
+import datetime
+import sys
+import argparse
+
+import json
+
+
+def clean_name(name):
+    """ Turn a name into a valid dot node name """
+    return name.replace('.','__').replace("-", "__")
+
+
+def trim_names(nodes):
+    """ Remove common prefixes from all graph names (DBT can be verbose!) """
+    names = [n['name'] for n in nodes.values()]
+    prefix = os.path.commonprefix(names)
+    if prefix:
+        for node in nodes.values():
+            node['name'] = node['name'].removeprefix(prefix)
```

Comment:
the removeprefix method does not work with anything earlier than 3.9.
Is it worth using a syntax compatible with other older versions here? Something like the below:
`node['name'] = node['name'][len(prefix):]`

Reply:
Replaced removeprefix as suggested -- good idea.

On the subject of google-cloud-bigquery dependency -- I just tried it out and all that's required is install google-cloud-dependency, but it also brings in a lot of other dependencies.

I think we can -
 - have a fully specified requirements.txt (I'd rather not..)
 - have just google-cloud-bigquery which will implicitly pull in dependencies (we'll need an exact version)
 - document in the README and in the script that google-cloud-bigquery needs to be installed.
 - capture the ModuleNotFoundError and document in the script that google-cloud-bigquery needs to be installed

What do you think about just doing the last two?

Reply:
yes, doing the last two sounds great

Reply:
Code updated with changes. Let me know.